src/server.py

```diff
@@ -1,8 +1,301 @@
+import re
+from datetime import datetime
+from typing import Union
+
 from fastapi import FastAPI
 
+from .data_fetcher import DataFetcher
+
+def current_year() -> str:
+    """Gets the current year, needed explicitly for the COURSE_DTL CoursePlanner API endpoint as other years break the response"""
+    return datetime.now().year
+
+def current_sem() -> str:
+    """Gets the current semester, to be used as a simple default for /courses/  """
+    return "Semester 1" if datetime.now().month <= 6 else "Semester 2" # Not the most bulletproof logic lol
+
+def meeting_date_convert(raw_date: str) -> dict[str]:
+    """Converts the date format given in the meetings to "MM-DD"
+
+    Args:
+        raw_date (str): The given meeting date in the format of "DD {3-char weekday} - DD {3-char weekday}"
+
+    Returns:
+        formatted_date (dict[str]): The formatted meeting date in the format of "MM-DD" in a dict containing 
+        the "start" and "end" keys and their corresponding dates
+    """
+    
+    months = ["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"]
+    start, end = raw_date.split(" - ")
+    
+    start_d, start_m = start.split()
+    start_m = str(months.index(start_m) + 1).zfill(2)
+    
+    end_d, end_m = end.split()
+    end_m = str(months.index(end_m) + 1).zfill(2)
+    
+    formatted_date = {
+        "start": f"{start_m}-{start_d.zfill(2)}",
+        "end": f"{end_m}-{end_d.zfill(2)}"
+    }
+    return formatted_date
+
+def meeting_time_convert(raw_time: str) -> str:
+    """Converts the time given in meetings to "HH:mm"
+
+    Args:
+        raw_time (str): The given meeting time in the format of "H{am/pm}"
+
+    Returns:
+        formatted_time (str): The formatted meeting time in the format of "HH:mm"
+    """
+    
+    period = raw_time[-2:]
+    hour = int(raw_time.replace(period, ""))
+    
+    if period.lower() == "pm":
+        hour += 12
+    
+    hour = str(hour).zfill(2)
+    formatted_time = f"{hour}:00"
+    return formatted_time
+
+def parse_requisites(raw_requisites: str) -> Union[dict[str], None]:
+    """Takes in a string of -requisites and returns a dict containing the original string, and a list of the parsed-out subjects
+
+    Args:
+        raw_requisites (str): The raw string containing a list of -requisites, usually in the format of 
+        "COMP SCI 1103, COMP SCI 2202, COMP SCI 2202B" as an example
+
+    Returns:
+        parsed_requisites (Union[dict[str], None]): Either a dict with "desc"s value being the original -requisites string and 
+        "subjects" being a list of the parsed -requisites, or None if raw_requisites is None
+    """
+    
+    if not raw_requisites:
+        return None
+    
+    # Note: This regex pattern was generated by Claude 3.5 Sonnet, works fine in my testing though
```

Comment:
We can maybe have tests (in another PR - an issue could be created for this) to ensure that this regex holds for several course names. Appreciate the honesty that it's generated using an LLM but the comment probably won't be necessary if we have sufficient tests (which is more bullet-proof than a disclaimer)

Reply:
I think Hugh started to add some basic tests in #6.